ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -609,24 +545,37 @@ private void analyzeHaving(final Node node) {
       analysis.setHavingExpression((Expression) node);
     }
 
-    private void addSelectItems(
-        final Optional<NodeLocation> location,
-        final LogicalSchema schema,
-        final String sourceAlias,
-        final String aliasPrefix
-    ) {
-      for (final Field field : schema.withoutAlias().valueFields()) {
+    private void visitSelectStar(final AllColumns allColumns) {
+
+      final Optional<NodeLocation> location = allColumns.getLocation();
+
+      final String prefix = allColumns.getPrefix()
+          .map(QualifiedName::toString)
+          .orElse("");
 
-        final QualifiedName name = QualifiedName.of(sourceAlias);
+      for (final AliasedDataSource source : analysis.getFromDataSources()) {
+
+        if (!prefix.isEmpty() && !prefix.equals(source.getAlias())) {
+          continue;
+        }
+
+        final QualifiedName name = QualifiedName.of(source.getAlias());
 
         final QualifiedNameReference nameRef = new QualifiedNameReference(location, name);
 
-        final DereferenceExpression selectItem =
-            new DereferenceExpression(location, nameRef, field.name());
+        final String aliasPrefix = analysis.isJoin()
+            ? source.getAlias() + "_"
+            : "";
+
+        for (final Field field : source.getDataSource().getSchema().fields()) {
```

Comment:
Because we were adding the value fields of the `DataSourceNode`s, which already have the ROWKEY and ROWTIME copied into the value schema. (See it's confusing!)